src/coreclr/nativeaot/System.Private.CoreLib/src/System/CrashInfo.cs

```diff
@@ -0,0 +1,389 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime;
+using System.Runtime.CompilerServices;
+using System.Text;
+using Internal.DeveloperExperience;
+
+namespace System
+{
+    internal unsafe class CrashInfo
+    {
+        /// <summary>
+        /// The kind or reason of crash for the triage JSON
+        /// </summary>
+        public enum CrashReason
+        {
+            Unknown = 0,
+            UnhandledException = 1,
+            EnvironmentFailFast = 2,
+            InternalFailFast = 3,
+        }
+
+        /// <summary>
+        /// Type of runtime (currently only NativeAOT is used)
+        /// </summary>
+        public enum RuntimeType
+        {
+            Unknown = 0,
+            Desktop = 1,
+            NetCore = 2,
+            SingleFile = 3,
+            NativeAOT = 4,
+        }
+
+        /// <summary>
+        /// This block is passed to the Watson DOTNET CLRMA provider and must not have any GC references
+        /// </summary>
+        private struct TriageBlockBuffer
+        {
+            public const int BufferSize = 8192;
+            public fixed byte Buffer[BufferSize];
+        }
+
+        private bool _comma;
+        private int _currentBufferIndex;
+        private int _reservedBuffer;
+        private static TriageBlockBuffer _triageBuffer;
```

Comment:
```suggestion
        private static TriageBlockBuffer s_triageBuffer;
```
Nit: Coding conventions

Reply:
It is meaningful to have the space pre-allocated as a static? We will do a bunch of allocations in the course of filling it in, so allocate the space itself on the GC heap would be ok too.

Reply:
It needs to be static so it is put into a "data" segment (R/W) so it is added to the dump (even full dumps for some reason).  This is probably also a Native AOT implementation detail but doesn't work if it is on the GC.

Reply:
Ok, this needs a comment that this code depends on this implementation detail.

Or even reserve the buffer in `C` and have a helper function that returns its address. It would provide stronger guarantee that the buffer lives in a physical data section.

Reply:
I like the C helper function returning the buffer.  What do you think of creating an export (like DotNetRuntimeDebugHeader) to this buffer for Linux? The Watson provider can lookup this export to find the crashinfo buffer.  Hand waving a lot of details, but we do need a Linux solution too.

Reply:
> What do you think of creating an export (like DotNetRuntimeDebugHeader) to this buffer for Linux?

Could it hang off the DotNetRuntimeDebugHeader? People do ask about the purpose of DotNetRuntimeDebugHeader so it would be nice not to introduce more things people will be bothered about enough to create issues in the repo.

Reply:
I'm going to do any Linux support in a separate PR after discussing the right approach here.